ofdb-core/src/usecases/review_places.rs

```diff
@@ -38,3 +40,45 @@ where
     );
     Ok(place_count)
 }
+
+#[derive(Debug, Error)]
+pub enum ReviewPlaceWithNonceError {
+    #[error(transparent)]
+    Repo(#[from] RepoError),
+    #[error("Invalid or outdated place revision")]
+    PlaceRevision,
+}
+
+pub fn review_place_with_nonce<R>(
+    repo: &R,
+    review_nonce: ReviewNonce,
+    new_status: ReviewStatus,
+) -> std::result::Result<(), ReviewPlaceWithNonceError>
+where
+    R: PlaceRepo,
+{
+    let ReviewNonce {
+        place_id,
+        place_revision,
+        ..
+    } = review_nonce;
+    let (place, _) = repo.get_place(place_id.as_str())?;
+
+    if place.revision != place_revision {
+        return Err(ReviewPlaceWithNonceError::PlaceRevision);
+    }
+
+    let activity = Activity::now(None);
+    log::info!("Changing review status of place {place_id} to {new_status:?}",);
```

Comment:
Logging the current revision and old status would be helpful for tracing issues.